npm/http/server/server.go

```diff
@@ -28,8 +28,9 @@ func NPMRestServerListenAndServe(config npmconfig.Config, npmEncoder json.Marsha
 
 	// prometheus handlers
 	if config.Toggles.EnablePrometheusMetrics {
-		rs.router.Handle(api.NodeMetricsPath, metrics.GetHandler(true))
-		rs.router.Handle(api.ClusterMetricsPath, metrics.GetHandler(false))
+		rs.router.Handle(api.NodeMetricsPath, metrics.GetHandler(metrics.CustomerNodeMetrics))
+		rs.router.Handle(api.ClusterMetricsPath, metrics.GetHandler(metrics.CustomerClusterMetrics))
+		rs.router.Handle(api.DataplaneHealthMetricsPath, metrics.GetHandler(metrics.DataplaneHealthMetrics))
```

Comment:
What is the reasoning behind 3 different endpoint URLs for scraping prometheus metrics? Traditionally it's just one endpoint serving metrics for all components of the workload.

Reply:
We already have two endpoints and have to keep this for back compat, so to keep in that same vein, I was thinking we leave customer-facing metrics on separate endpoints (nodemetrics and clustermetrics) and internal on others

Reply:
we could rename dataplane health to "daemon-metrics", "internal-metrics", or just "metrics". Wdyt?

Reply:
option 2: we can dump all dataplane health metrics onto node-metrics

We would probably want to update Container Insights to drop those num_failures metrics then though (they're set up to scrape all metrics from "/node-metrics"

Reply:
@huntergregory  lets think the future, this distinction we add now may live forever for the same backward compatibility reason. so lets think about next 2-3 year plan, what is the best course forward. I would not want to have multiple different endpoints for different kinds of metrics. All the "internal" metrics are relevant for customers too and they would want to see them in their workbooks. What if we keep existing metrics as is and just distribute the new metrics onto node metrics for daemon specifc and cluster metrics for generic metrics ?

Reply:
Good points. Let's go with this strategy for the daemon. 

Not necessary to decide now, but I'm thinking we should just have a "/metrics" endpoint for the fan-out controller, and those metrics should have their own registry and initialization function.

Reply:
Final update on this:
Now we have controller metrics and daemon metrics. Seems easiest to preserve the /node-metrics and /cluster-metrics endpoints for both the fan-out and vanilla npm, since both the vanilla controller and daemon have metrics on each endpoint.